pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symboltable/ScopeAndDeclarationFinder.java

```diff
@@ -304,13 +303,6 @@ public Object visit(ASTForStatement node, Object data) {
         return data;
     }
 
-    @Override
-    public Object visit(ASTIfStatement node, Object data) {
```

Comment:
Don't `if` statements have their own scope?

Reply:
No, they don't do on their own. Only if using a block after the if (in which case the block itself defines the scope). I was just writing a brief explanation of the situation before / after this change, please review it and let me know if you have doubts